web/src/layout/main.jsx

```diff
@@ -150,13 +152,20 @@ const Main = ({ electricityMixMode, hasConnectionWarning }) => {
             )}
           </ErrorBoundary>
 
-          <div id="connection-warning" className={`flash-message ${hasConnectionWarning ? 'active' : ''}`}>
+          <div id="connection-warning" className={`flash-message ${failedRequestType ? 'active' : ''}`}>
             <div className="inner">
               {__('misc.oops')}{' '}
               <button
                 type="button"
                 onClick={(e) => {
-                  dispatch(GRID_DATA_FETCH_REQUESTED());
+                  if (failedRequestType === 'grid') {
+                    dispatch(GRID_DATA_FETCH_REQUESTED({ features, selectedTimeAggregate }));
+                  }
+                  if (failedRequestType === 'zone') {
+                    dispatch(
+                      ZONE_HISTORY_FETCH_REQUESTED({ zoneId: failedRequestZoneId, features, selectedTimeAggregate })
+                    );
+                  }
```

Comment:
Since we are adding a bit more logic to this component, I'd perhaps move it to its own component eg <ConnectionWarning />